### ISSUE TITLE
move-form has the possibility of creating invalid namespaces

### ISSUE THREAD

**Issue by Malabarba** [clojure-emacs/clj-refactor.el#199]
If you have the files

    (ns website.views.analyse-repo
      (:require [hiccup.core :as h]))
    
    (defn index [name]
      (h/h name))

and

    (ns website.views.layout
      (:require [hiccup.page :as h]))
    
    (defn other-func []
      (h/html5 "OK"))

and you use move-form to move `index` to `layout.clj`, then the resulting namespace in that file will be 

    (ns website.views.layout
      (:require [hiccup.page :as h]
                [hiccup.core :as h]))

Maybe clj-refactor should offer to fix that when it happens? 

However, that sounds like it's probably very troublesome to implement. If that's the case, it would be nice to at least get a message about that before or after I make the move, instead of finding out with an evaluation error further down the line.

**Comment by expez** [comment]
I think letting the user know that manual intervention is required is the way to go.  We'd have to make too many assumptions to fix this manually, and if we have to prompt the user at all those decision points it's likely to be just as fast for the user to fix it manually.  

This refactoring has been around for almost 2 years now and you're the first one to file a bug about this problem, which is another argument for not expending too much effort on this.

**Comment by magnars** [comment]
Yeah `move-form` is quite naive in its implementation. I'm afraid it would have to go through a major overhaul to be a solid error-free operation. Think `remove-requires` vs `clean-ns`.

**Comment by Malabarba** [comment]
> This refactoring has been around for almost 2 years now and you're the first one to file a bug about this problem, which is another argument for not expending too much effort on this.

Agreed. :)

**Comment by MalloZup** [comment]
autogenerated with https://github.com/MalloZup/doghub:  issue inactive since 450 days. Please update the issue or close it
